### DISCUSSION TITLE
How to setup a development enviroment for netbox?

### DISCUSSION THREAD

**Discussion by yoloknight** [netbox-community/netbox#14080] [Q&A] [answered]
Hi,
I wanted to develop in netbox but I cant find a tutorial to setup my development enviroment.
I would like to use "Virtual Studio Code" and "Windows". So far as I see it are serveral netbox components only for Linux, like Redis. 
So for Redis I have to use something like "WSL" for it to work on Windows.  For Postgres I have to use something like "pgadmin".

So I wanted to ask is there a "step by step" guide how I can get a netbox development enviroment running? Like for the steps from downloading the netbox project from github, configure the "configuration.py", preparing the Postgres DB and Redis Instance. Until I have the netbox development instance running for "scripting" and "plugin" programming.

thanks and best regards

**Comment by raphis** [comment]
I use docker-compose which has a pretty simple startup-script for netbox (also under windows). This is mentioned here: https://github.com/netbox-community/netbox-docker

You can then just import the database (and if you want you can also copy media data like device-type pictures) and off you go with a local running copy of your netbox.

**Comment by yoloknight** [comment, marked as answer]
Ok I found a solution for my problem.

This are the steps:
1. Use WSL and activate systemd
2. Use nginx with proxy_pass and proxy_redirect for your localhost

My problem was more a nginx problem because you need a double proxy configuration.
